tests/test_integration.py

```diff
@@ -61,7 +61,14 @@ def test_pypi_upload(sampleproject_dist):
         sampleproject_token,
         str(sampleproject_dist),
     ]
-    cli.dispatch(command)
+
+    try:
+        cli.dispatch(command)
+    except requests.HTTPError as exc:
+        # Workaround intermittent failure from testpypi
+        if exc.response.status_code >= 500:
+            pytest.xfail(f"{exc.response.status_code} from {exc.response.url}")
+        raise
```

Comment:
Following up on https://github.com/pypa/twine/pull/686#issuecomment-678715478.

You can see this (and the similar workaround below) in action at https://travis-ci.com/github/pypa/twine/jobs/378434147#L306-L309:

```
=========================== short test summary info ============================
XFAIL tests/test_integration.py::test_pypi_upload
  reason: 502 from https://test.pypi.org/legacy/
XFAIL tests/test_integration.py::test_pypi_error
  reason: HTTPError: 502 Bad Gateway from https://test.pypi.org/legacy/
Bad Gateway
============ 155 passed, 2 xfailed, 8 warnings in 112.77s (0:01:52) ============
```

It also let the GitHub Action pass:

https://github.com/pypa/twine/runs/1037251938?check_suite_focus=true#step:5:59
https://github.com/pypa/twine/runs/1037252012?check_suite_focus=true#step:5:59

Unfortunately, the [Travis build](https://travis-ci.com/github/pypa/twine/builds/181661132) still failed, due to a related failure in the [docs build](https://travis-ci.com/github/pypa/twine/jobs/378434153#L479-L480):

```
Warning, treated as error:
/home/travis/build/pypa/twine/docs/contributing.rst.rst:190:broken link: https://test.pypi.org/manage/project/twine/collaboration/ (502 Server Error: Bad Gateway for url: https://test.pypi.org/manage/project/twine/collaboration/)
```

I don't love this "conditional `xfail`" hack. Open to suggestions (including "we can live with TestPyPI being flaky").

Reply:
@di and @jaraco Any thoughts on this?

I'd at least like to merge the "rerun failures" functionality from https://github.com/pypa/twine/pull/686/commits/82571d0e382f82467066e31638f2c5dce1976f34.

I'm on the fence about the conditional `xfail` in https://github.com/pypa/twine/pull/686/commits/3d767f5f6182473e4e2c602fad975ce07e2c19d7.